Metrics/start-windows-build.ps1

```diff
@@ -0,0 +1,365 @@
+Param(
+    [Parameter(Mandatory=$false)]
+    [string]$GitURL="https://github.com/dcos/dcos-metrics", 
+    [Parameter(Mandatory=$false)]
+    [string]$Branch="master",
+    [Parameter(Mandatory=$false)]
+    [string]$CommitID,
+    [Parameter(Mandatory=$false)]
+    [string]$ParametersFile="${env:WORKSPACE}\build-parameters.json"
+)
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+$ErrorActionPreference = "Stop"
+
+$globalVariables = (Resolve-Path "$PSScriptRoot\..\global-variables.ps1").Path
+$ciUtils = (Resolve-Path "$PSScriptRoot\..\Modules\CIUtils").Path
+
+Import-Module $ciUtils
+. $globalVariables
+
+$global:PARAMETERS = @{
+    "BUILD_STATUS" = $null
+    "LOGS_URLS" = @()
+    "FAILED_COMMAND" = $null
+}
+function Install-Prerequisites {
+    $prerequisites = @{
+        'git'= @{
+            'url'= $GIT_URL
+            'install_args' = @("/SILENT")
+            'install_dir' = $GIT_DIR
+        }
+        'go'= @{
+            'url'= $GOLANG_URL_1_94
+            'install_args'= @("/quiet")
+            'install_dir'= $GOLANG_DIR
+        }
+        '7zip'= @{
+            'url'= $7ZIP_URL
+            'install_args'= @("/q")
+            'install_dir'= $7ZIP_DIR
+        }
+    }
+    foreach($program in $prerequisites.Keys) {
+        if(Test-Path $prerequisites[$program]['install_dir']) {
+            Write-Output "$program is already installed"
+            continue
+        }
+        Write-Output "Downloading $program from $($prerequisites[$program]['url'])"
+        $fileName = $prerequisites[$program]['url'].Split('/')[-1]
+        $programFile = Join-Path $env:TEMP $fileName
+
+        Start-ExecuteWithRetry { Invoke-WebRequest -Uri $prerequisites[$program]['url'] -OutFile $programFile}
+        $parameters = @{
+            'FilePath' = $programFile
+            'ArgumentList' = $prerequisites[$program]['install_args']
+            'Wait' = $true
+            'PassThru' = $true
+        }
+        if($programFile.EndsWith('.msi')) {
+            $parameters['FilePath'] = 'msiexec.exe'
+            $parameters['ArgumentList'] += @("/i", $programFile)
+        }
+        Write-Output "Installing $programFile"
+        $p = Start-Process @parameters
+        if($p.ExitCode -ne 0) {
+            Throw "Failed to install prerequisite $programFile during the environment setup : $($p.ExitCode)"
+        }
+    }
+    # Add all the tools to PATH
+    $toolsDirs = @("$GOLANG_DIR\bin", "$GIT_DIR\cmd", "$GIT_DIR\bin", "$7ZIP_DIR")
+    $env:PATH += ';' + ($toolsDirs -join ';')
+    
+    $env:GOPATH = $METRICS_GOPTH_DIR
```

Comment:
I would suggesting moving this `$env:GOPATH` and `$env:GOBIN` sets in the `New-TestingEnvironment` function.

Reply:
Good suggestion